### PR TITLE
feat(cmd): respect NO_COLOR and add a --no-color flag

### LINKED ISSUE #309: Respect NO_COLOR and add a --no-color flag
## Problem

gup always emits ANSI colors via `fatih/color` and has no way to disable them
other than what the library auto-detects. There is no support for the de-facto
standard `NO_COLOR` environment variable, nor an explicit `--no-color` flag.
This hurts output when piping/redirecting, in CI logs, and for users who set
`NO_COLOR` globally.

(Note: `--quiet` / summary output is tracked separately in #290; this issue is
specifically about color control.)

## Proposal

- Honor the `NO_COLOR` environment variable (disable color when it is set to
  any non-empty value), following https://no-color.org/.
- Add an explicit `--no-color` flag that forces color off regardless of TTY
  detection.
- Keep current auto-detected colored output as the default for interactive
  terminals.

## Acceptance Criteria

- Setting `NO_COLOR` disables colored output.
- `--no-color` disables colored output explicitly.
- Behavior is documented in help text and README.
- Tests cover the env-var and flag paths.

### PR DESCRIPTION
**Pull request by nao1215**
## Summary

gup colorized its output via fatih/color with no explicit way to turn colors off, which hurts piping/redirecting, CI logs, and users who set NO_COLOR globally. This adds a persistent `--no-color` flag and honors the de-facto-standard `NO_COLOR` environment variable, while keeping the existing automatic non-TTY color detection as the default. Closes #309.

## Changes

- `cmd/flags.go`: add the persistent `--no-color` flag (`addNoColorFlag`) and `applyColorPreference`, which sets `color.NoColor = true` when `--no-color` is passed or `NO_COLOR` is non-empty.
- `cmd/root.go`: register the flag and apply the preference from a root `PersistentPreRunE`, so every subcommand honors it.
- `README.md`: flip the `NO_COLOR support` cell in the feature comparison from `No` to `Yes` and add a short "Disable colorized output" usage subsection.
- `cmd/testdata/bug_report/bug_report.txt`: update the `--help` golden file to include the new `Global Flags` section.
- `cmd/flags_test.go`: cover both paths — a table test for `applyColorPreference` (flag, non-empty/empty `NO_COLOR`) and an end-to-end test running `version --no-color` through the root command.

## Design Decisions

`applyColorPreference` only ever sets `color.NoColor = true` and never back to `false`, so it layers on top of fatih/color's startup non-TTY/`NO_COLOR` detection instead of overriding it; this keeps "no color when piping" working and avoids forcing color on. The `NO_COLOR` check uses `os.Getenv("NO_COLOR") != ""` to match both the no-color.org convention and fatih/color's own `noColorIsSet`, so an empty value does not disable color. The flag is a single persistent flag on the root command applied via `PersistentPreRunE`, because no subcommand defines its own persistent pre-run, so one definition covers all of them (`update`, `check`, `list`, `remove`, `import`, `migrate`, `export`, `version`, `bug-report`, `man`). Color is also re-checked at runtime rather than relying on fatih/color's import-time evaluation, which makes the `NO_COLOR` path testable and robust.

## Limitations

The bare `gup` command (no subcommand) is not runnable and shows only help, so its `PersistentPreRunE` does not run; this is harmless because help output is uncolored. The English README is updated here, but the six translated READMEs under `doc/` are intentionally left for the separate translation-sync effort tracked in #306 (they already lag the English source by whole sections), rather than shipping unverified machine translations of the new subsection.

## Summary by CodeRabbit

* **New Features**
  * Added `--no-color` flag to disable colorized output across all commands
  * Color output can also be disabled via the `NO_COLOR` environment variable
  * Updated documentation with usage examples and feature comparison updates

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -321,6 +321,13 @@ $ gup update --notify
 ![success](./doc/img/notify_success.png)
 ![warning](./doc/img/notify_warning.png)
 
+### Disable colorized output
+gup colorizes its output by default. To turn colors off, pass `--no-color` or set the `NO_COLOR` environment variable to a non-empty value (following the [NO_COLOR](https://no-color.org/) convention). This is useful when piping output, in CI logs, or with `NO_COLOR` set globally.
+```shell
+$ gup update --no-color
+$ NO_COLOR=1 gup update
```
[the remark this answers is not in the document]
Comment (nao1215) [reply README.md:328]: Thanks, but I'll keep the `$` prefix here for consistency: the README already uses `$ <command>` in 24 existing shell examples (e.g. `$ gup update --notify`, `$ gup check`), so dropping it only in this new block would make it the odd one out. MD014 is also not enforced in this repo — there's no markdownlint config, and the reviewdog workflow runs only misspell and actionlint — so this is a style preference rather than a CI gate. If we ever want bare (copy-paste-friendly) commands, that's better done as a separate sweep over all 24 examples.
